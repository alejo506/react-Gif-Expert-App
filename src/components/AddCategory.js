import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategorias }) => {

  const [inputValue, setInputValue] = useState(''); //Si no se pone el string vacio en el use a un error de que el componente está cambiando incontroladamente, este error es debido a que si no se pone nada el valor es undefined por ende el valor del input value es undefined

  //Con esta función lo que se quiere hacer es que se escriba algo en la caja de texto y se presione para que muestre algo
  const handleSubmit = (e) => {

    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategorias((categ) => [inputValue, ...categ]);
      setInputValue("");
    }
    // console.log("Submit hecho");
  };

  // Con esta función ya se puede cambiar el valor dentro del input. 
  const handleInputChance = (e) => {


    setInputValue(e.target.value);
  };
  return (

    (
      <form onSubmit={handleSubmit}>
        <input
          //Atributos
          className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search"
          value={inputValue}
          onChange={handleInputChance}
        />
      </form>
    )
  );

};

AddCategory.propTypes = {


  setCategorias: PropTypes.func.isRequired,
};
