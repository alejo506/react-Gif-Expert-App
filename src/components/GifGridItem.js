
import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ cod, title, url }) => { //Recibe las propiedades de las imagenes de GifGrid //El cod si quiere se puede quitar porque no se está usando acá

    // console.log(cod, title, url);
    return (
        <div className="card animate__animated animate__flipInX">
            <img src={url} className="card-img-top" alt={title} />
            <div className="card-body">
                <h5 className="card-title">{title}</h5>
            </div>
        </div>
    )
}

// TAREA. Seccion# 8, Video 88
// 1. Enzime. (Instalar desde la pagina). npm i --save-dev enzyme enzyme-adapter-react-16
// 2. Enzime to JSON
// 3. Debe mostrar el componente correctamente
//     A)Shallow
//     b)Wrapper
//     c)Wrapper .toMatchSnapshot()

// Seccion#8, video 89
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
