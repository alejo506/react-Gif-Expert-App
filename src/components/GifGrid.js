import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import { Spinner } from 'react-awesome-spinners';


export const GifGrid = ({ categ }) => {// 1.Recibe la categoría como argumento

    //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

    // data: images es para renombrar data, data ahora es images, se puede utlizar data pero es bueno saber que para renombrar en la desestructuración agregamos el nombre quetiene actualmente ":" seguido del nuevo nombre que queremos asignar a esa propiedad, por lo que images es el nuevo nombre de data.
    //Hace el llamado a la función useFetchGifs(categ) por lo que se redirige al archivo useFetchGifs.js
    const { data: images, loading } = useFetchGifs(categ); //El categ lo estoy enviando a useFetchGifs.js, para hacer uso de ella en dicho archivo


    //---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

    return (
        <>
            <h2> {categ} </h2>

            {loading && <Spinner />}  {/*En lugar de utilizar un operador ternario, utilizamos un operador And, por lo que sería como 
            decir si loading===true entonces ponga el Spinner, de lo contrario no ponga nada, pero no debemos especificar que no ponga 
            nada en cambio con el operador ternario si tendriamos que poner null para que no muestre nada, por lo que es una forma más corta utilizar &&*/}
            {/* El Spinner lo agregué por mi cuenta, hice la importación respectiva para utlizarlo*/ }
            
            <div className="card-grid">

                {
                    //Aquí en lugar de "data" ahora utilizamos el nuevo nombre "images"
                    images.map(img => (
                        <GifGridItem
                            key={img.cod}// Siempre se agrega el key para que no presente error
                            {...img}//Envíe un objeto con las propiedades independientes id, title , url
                        />

                    ))
                }
            </div>
        </>
    )
}
