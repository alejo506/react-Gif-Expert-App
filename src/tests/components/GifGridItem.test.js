// Con <GifGridItem/>  ya estaríamos utilizando React o JSX por eso importamos React
import React from 'react';
import { shalow, shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem />', () => {
    test('debe mostrar el componente correctamente', () => {
        
        // Seccion 8, video 89
        const title = 'Un título';
        const url = 'htpps://localhost/algo.jpg';

        // <GifGridItem/> es el componente al cual le vamos a hacer pruebas 
        const wrapper = shallow( <GifGridItem title= { title } url= { url }/> )
        
        expect( wrapper ).toMatchSnapshot(); //Creamos un Snapshot
    })
    
    
})
