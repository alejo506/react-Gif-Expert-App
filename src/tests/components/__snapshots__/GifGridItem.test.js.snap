// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="Un título"
    className="card-img-top"
    src="htpps://localhost/algo.jpg"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Un título
    </h5>
  </div>
</div>
`;
